[
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "<!--e57a25ab6845829454e8d69fc972939a-->\n\nThe following sections might be updated with supplementary metadata relevant to reviewers and maintainers.\n\n<!--006a51241073e994b41acfe9ec718e94-->\n### Code Coverage\nFor detailed information about the code coverage, see the [test coverage report](https://corecheck.dev/bitcoin/bitcoin/pulls/29498).\n<!--021abf342d371248e50ceaed478a90ca-->\n### Reviews\nSee [the guideline](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#code-review) for information on the review process.\n| Type | Reviewers |\n| ---- | --------- |\n| Stale ACK | [maflcko](https://github.com/bitcoin/bitcoin/pull/29498#issuecomment-1968487248) |\n\nIf your review is incorrectly listed, please react with ð to this comment and the bot will ignore it on the next update.\n",
      "created_at" : "2024-02-28T00:47:34Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#issuecomment-1967982197",
      "id" : 1967982197,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29498",
      "node_id" : "IC_kwDOABII5851TQZ1",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1967982197/reactions"
      },
      "updated_at" : "2024-03-12T22:55:40Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1967982197",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/39886733?v=4",
         "events_url" : "https://api.github.com/users/DrahtBot/events{/privacy}",
         "followers_url" : "https://api.github.com/users/DrahtBot/followers",
         "following_url" : "https://api.github.com/users/DrahtBot/following{/other_user}",
         "gists_url" : "https://api.github.com/users/DrahtBot/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/DrahtBot",
         "id" : 39886733,
         "login" : "DrahtBot",
         "node_id" : "MDQ6VXNlcjM5ODg2NzMz",
         "organizations_url" : "https://api.github.com/users/DrahtBot/orgs",
         "received_events_url" : "https://api.github.com/users/DrahtBot/received_events",
         "repos_url" : "https://api.github.com/users/DrahtBot/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/DrahtBot/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/DrahtBot/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/DrahtBot"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "Might you be able to modify the message as well? It will be the most available info when people encounter it.",
      "created_at" : "2024-02-28T07:05:35Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#issuecomment-1968357112",
      "id" : 1968357112,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29498",
      "node_id" : "IC_kwDOABII5851Ur74",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1968357112/reactions"
      },
      "updated_at" : "2024-02-28T07:05:35Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1968357112",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/5470?v=4",
         "events_url" : "https://api.github.com/users/Empact/events{/privacy}",
         "followers_url" : "https://api.github.com/users/Empact/followers",
         "following_url" : "https://api.github.com/users/Empact/following{/other_user}",
         "gists_url" : "https://api.github.com/users/Empact/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/Empact",
         "id" : 5470,
         "login" : "Empact",
         "node_id" : "MDQ6VXNlcjU0NzA=",
         "organizations_url" : "https://api.github.com/users/Empact/orgs",
         "received_events_url" : "https://api.github.com/users/Empact/received_events",
         "repos_url" : "https://api.github.com/users/Empact/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/Empact/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/Empact/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/Empact"
      }
   },
   {
      "author_association" : "MEMBER",
      "body" : "lgtm ACK d4e36ae80d4b3f03647fd9057461edf7ecd794a3",
      "created_at" : "2024-02-28T08:42:23Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#issuecomment-1968487248",
      "id" : 1968487248,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29498",
      "node_id" : "IC_kwDOABII5851VLtQ",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1968487248/reactions"
      },
      "updated_at" : "2024-02-28T08:42:23Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1968487248",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/6399679?v=4",
         "events_url" : "https://api.github.com/users/maflcko/events{/privacy}",
         "followers_url" : "https://api.github.com/users/maflcko/followers",
         "following_url" : "https://api.github.com/users/maflcko/following{/other_user}",
         "gists_url" : "https://api.github.com/users/maflcko/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/maflcko",
         "id" : 6399679,
         "login" : "maflcko",
         "node_id" : "MDQ6VXNlcjYzOTk2Nzk=",
         "organizations_url" : "https://api.github.com/users/maflcko/orgs",
         "received_events_url" : "https://api.github.com/users/maflcko/received_events",
         "repos_url" : "https://api.github.com/users/maflcko/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/maflcko/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/maflcko/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/maflcko"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "Updated in [be027e0](https://github.com/bitcoin/bitcoin/pull/29498/commits/be027e00f1c6d7c59790d886d280b570f848eae1) I added a line to enable logging and it now gives this error which it did not before because it would throw an error because logging was not enabled\r\n\r\nNew Error message\r\n```\r\n./test/functional/feature_bind_port_discover.py --tmpdir=./temp\r\n2024-03-12T23:08:06.633000Z TestFramework (ERROR): Unexpected exception caught during testing\r\nTraceback (most recent call last):\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 130, in main\r\n    self.setup()\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 267, in setup\r\n    os.makedirs(self.options.tmpdir, exist_ok=False)\r\n  File \"<frozen os>\", line 225, in makedirs\r\nFileExistsError: [Errno 17] File exists: '/Users/kevkevin/DEVDIR/bitcoin/temp'\r\nTraceback (most recent call last):\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/./test/functional/feature_bind_port_discover.py\", line 78, in <module>\r\n    BindPortDiscoverTest().main()\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 154, in main\r\n    exit_code = self.shutdown()\r\n                ^^^^^^^^^^^^^^^\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 310, in shutdown\r\n    self.network_thread.close()\r\n    ^^^^^^^^^^^^^^^^^^^^^^^^^\r\nAttributeError: 'NoneType' object has no attribute 'close'\r\n```\r\n\r\nPrevious Error message\r\n```\r\nTraceback (most recent call last):\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 130, in main\r\n    self.setup()\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 266, in setup\r\n    os.makedirs(self.options.tmpdir, exist_ok=False)\r\n  File \"<frozen os>\", line 225, in makedirs\r\nFileExistsError: [Errno 17] File exists: '/Users/kevkevin/DEVDIR/bitcoin/temp'\r\n\r\nDuring handling of the above exception, another exception occurred:\r\n\r\nTraceback (most recent call last):\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 148, in main\r\n    self.log.exception(\"Unexpected exception caught during testing\")\r\n    ^^^^^^^^\r\nAttributeError: 'BindPortDiscoverTest' object has no attribute 'log'\r\n\r\nDuring handling of the above exception, another exception occurred:\r\n\r\nTraceback (most recent call last):\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/./test/functional/feature_bind_port_discover.py\", line 78, in <module>\r\n    BindPortDiscoverTest().main()\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 154, in main\r\n    exit_code = self.shutdown()\r\n                ^^^^^^^^^^^^^^^\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 304, in shutdown\r\n    if self.success == TestStatus.FAILED and self.options.pdbonfailure:\r\n       ^^^^^^^^^^^^\r\nAttributeError: 'BindPortDiscoverTest' object has no attribute 'success'\r\n```",
      "created_at" : "2024-03-12T23:10:15Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#issuecomment-1992721164",
      "id" : 1992721164,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29498",
      "node_id" : "IC_kwDOABII5852xoMM",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1992721164/reactions"
      },
      "updated_at" : "2024-03-12T23:10:15Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1992721164",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/15950706?v=4",
         "events_url" : "https://api.github.com/users/kevkevinpal/events{/privacy}",
         "followers_url" : "https://api.github.com/users/kevkevinpal/followers",
         "following_url" : "https://api.github.com/users/kevkevinpal/following{/other_user}",
         "gists_url" : "https://api.github.com/users/kevkevinpal/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/kevkevinpal",
         "id" : 15950706,
         "login" : "kevkevinpal",
         "node_id" : "MDQ6VXNlcjE1OTUwNzA2",
         "organizations_url" : "https://api.github.com/users/kevkevinpal/orgs",
         "received_events_url" : "https://api.github.com/users/kevkevinpal/received_events",
         "repos_url" : "https://api.github.com/users/kevkevinpal/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/kevkevinpal/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/kevkevinpal/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/kevkevinpal"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "I may want to find a way to remove the logs after `FileExistsError: [Errno 17] File exists: '/Users/kevkevin/DEVDIR/bitcoin/temp'` to make this error more clear",
      "created_at" : "2024-03-12T23:10:57Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#issuecomment-1992721675",
      "id" : 1992721675,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29498",
      "node_id" : "IC_kwDOABII5852xoUL",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1992721675/reactions"
      },
      "updated_at" : "2024-03-12T23:10:57Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1992721675",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/15950706?v=4",
         "events_url" : "https://api.github.com/users/kevkevinpal/events{/privacy}",
         "followers_url" : "https://api.github.com/users/kevkevinpal/followers",
         "following_url" : "https://api.github.com/users/kevkevinpal/following{/other_user}",
         "gists_url" : "https://api.github.com/users/kevkevinpal/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/kevkevinpal",
         "id" : 15950706,
         "login" : "kevkevinpal",
         "node_id" : "MDQ6VXNlcjE1OTUwNzA2",
         "organizations_url" : "https://api.github.com/users/kevkevinpal/orgs",
         "received_events_url" : "https://api.github.com/users/kevkevinpal/received_events",
         "repos_url" : "https://api.github.com/users/kevkevinpal/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/kevkevinpal/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/kevkevinpal/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/kevkevinpal"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "force pushed once more [eb28434](https://github.com/bitcoin/bitcoin/pull/29498/commits/eb284341b266d92b7890ba879b2c2b9271137d08)\r\n\r\nthe error log now makes sense\r\n\r\n```\r\n./test/functional/feature_bind_port_discover.py --tmpdir=./temp\r\n2024-03-12T23:12:18.583000Z TestFramework (ERROR): Unexpected exception caught during testing\r\nTraceback (most recent call last):\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 130, in main\r\n    self.setup()\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 267, in setup\r\n    os.makedirs(self.options.tmpdir, exist_ok=False)\r\n  File \"<frozen os>\", line 225, in makedirs\r\nFileExistsError: [Errno 17] File exists: '/Users/kevkevin/DEVDIR/bitcoin/temp'\r\n2024-03-12T23:12:18.584000Z TestFramework (INFO): Stopping nodes\r\n2024-03-12T23:12:18.584000Z TestFramework (WARNING): Not cleaning up dir /Users/kevkevin/DEVDIR/bitcoin/temp\r\n2024-03-12T23:12:18.584000Z TestFramework (ERROR): Test failed. Test logging available at /Users/kevkevin/DEVDIR/bitcoin/temp/test_framework.log\r\n2024-03-12T23:12:18.584000Z TestFramework (ERROR):\r\n2024-03-12T23:12:18.584000Z TestFramework (ERROR): Hint: Call /Users/kevkevin/DEVDIR/bitcoin/test/functional/combine_logs.py '/Users/kevkevin/DEVDIR/bitcoin/temp' to consolidate all logs\r\n2024-03-12T23:12:18.584000Z TestFramework (ERROR):\r\n2024-03-12T23:12:18.584000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.\r\n2024-03-12T23:12:18.585000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues\r\n2024-03-12T23:12:18.585000Z TestFramework (ERROR):\r\n```",
      "created_at" : "2024-03-12T23:14:45Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#issuecomment-1992724797",
      "id" : 1992724797,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29498",
      "node_id" : "IC_kwDOABII5852xpE9",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1992724797/reactions"
      },
      "updated_at" : "2024-03-12T23:14:45Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1992724797",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/15950706?v=4",
         "events_url" : "https://api.github.com/users/kevkevinpal/events{/privacy}",
         "followers_url" : "https://api.github.com/users/kevkevinpal/followers",
         "following_url" : "https://api.github.com/users/kevkevinpal/following{/other_user}",
         "gists_url" : "https://api.github.com/users/kevkevinpal/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/kevkevinpal",
         "id" : 15950706,
         "login" : "kevkevinpal",
         "node_id" : "MDQ6VXNlcjE1OTUwNzA2",
         "organizations_url" : "https://api.github.com/users/kevkevinpal/orgs",
         "received_events_url" : "https://api.github.com/users/kevkevinpal/received_events",
         "repos_url" : "https://api.github.com/users/kevkevinpal/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/kevkevinpal/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/kevkevinpal/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/kevkevinpal"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "new error log in c5eb4688a90577c573861cc56f10c9dac15f97de\r\n\r\nlet me know if you want to change the verbiage and if the dir exists we use the default location and then raise and exception\r\n```\r\n./test/functional/p2p_tx_privacy.py --tmpdir=./tmp\r\n2024-03-13T01:48:26.199000Z TestFramework (ERROR): The tmpdir must not already exist ./tmp\r\n2024-03-13T01:48:26.200000Z TestFramework (ERROR): Unexpected exception caught during testing\r\nTraceback (most recent call last):\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 130, in main\r\n    self.setup()\r\n  File \"/Users/kevkevin/DEVDIR/bitcoin/test/functional/test_framework/test_framework.py\", line 268, in setup\r\n    os.makedirs(self.options.tmpdir, exist_ok=False)\r\n  File \"<frozen os>\", line 225, in makedirs\r\nFileExistsError: [Errno 17] File exists: '/Users/kevkevin/DEVDIR/bitcoin/tmp'\r\n2024-03-13T01:48:26.201000Z TestFramework (INFO): Stopping nodes\r\n2024-03-13T01:48:26.201000Z TestFramework (WARNING): Not cleaning up dir /var/folders/9g/cvx014yx4dq5lwl_x5zvn_j80000gn/T/bitcoin_func_test_9s71cl9w\r\n2024-03-13T01:48:26.201000Z TestFramework (ERROR): Test failed. Test logging available at /var/folders/9g/cvx014yx4dq5lwl_x5zvn_j80000gn/T/bitcoin_func_test_9s71cl9w/test_framework.log\r\n2024-03-13T01:48:26.201000Z TestFramework (ERROR):\r\n2024-03-13T01:48:26.201000Z TestFramework (ERROR): Hint: Call /Users/kevkevin/DEVDIR/bitcoin/test/functional/combine_logs.py '/var/folders/9g/cvx014yx4dq5lwl_x5zvn_j80000gn/T/bitcoin_func_test_9s71cl9w' to consolidate all logs\r\n2024-03-13T01:48:26.201000Z TestFramework (ERROR):\r\n2024-03-13T01:48:26.201000Z TestFramework (ERROR): If this failure happened unexpectedly or intermittently, please file a bug and provide a link or upload of the combined log.\r\n2024-03-13T01:48:26.201000Z TestFramework (ERROR): https://github.com/bitcoin/bitcoin/issues\r\n2024-03-13T01:48:26.202000Z TestFramework (ERROR):\r\n```",
      "created_at" : "2024-03-13T01:50:01Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#issuecomment-1993075076",
      "id" : 1993075076,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/29498",
      "node_id" : "IC_kwDOABII5852y-mE",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1993075076/reactions"
      },
      "updated_at" : "2024-03-13T01:50:01Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1993075076",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/15950706?v=4",
         "events_url" : "https://api.github.com/users/kevkevinpal/events{/privacy}",
         "followers_url" : "https://api.github.com/users/kevkevinpal/followers",
         "following_url" : "https://api.github.com/users/kevkevinpal/following{/other_user}",
         "gists_url" : "https://api.github.com/users/kevkevinpal/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/kevkevinpal",
         "id" : 15950706,
         "login" : "kevkevinpal",
         "node_id" : "MDQ6VXNlcjE1OTUwNzA2",
         "organizations_url" : "https://api.github.com/users/kevkevinpal/orgs",
         "received_events_url" : "https://api.github.com/users/kevkevinpal/received_events",
         "repos_url" : "https://api.github.com/users/kevkevinpal/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/kevkevinpal/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/kevkevinpal/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/kevkevinpal"
      }
   },
   {
      "_links" : {
         "html" : {
            "href" : "https://github.com/bitcoin/bitcoin/pull/29498#discussion_r1522713384"
         },
         "pull_request" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/29498"
         },
         "self" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1522713384"
         }
      },
      "author_association" : "MEMBER",
      "body" : "Not sure about using a different tempdir than the one the user asked for as fallback to start logging. \r\n\r\nI don't think code needs to be written for users that fail to read the docs, and also fail to read the python stacktrace?",
      "commit_id" : "d4e36ae80d4b3f03647fd9057461edf7ecd794a3",
      "created_at" : "2024-03-13T08:07:58Z",
      "diff_hunk" : "@@ -262,8 +262,15 @@ def setup(self):\n \n         # Set up temp directory and start logging\n         if self.options.tmpdir:\n-            self.options.tmpdir = os.path.abspath(self.options.tmpdir)\n-            os.makedirs(self.options.tmpdir, exist_ok=False)\n+            tmpdirpath = self.options.tmpdir\n+            try:\n+                self.options.tmpdir = os.path.abspath(tmpdirpath)\n+                os.makedirs(self.options.tmpdir, exist_ok=False)\n+            except Exception:\n+                self.options.tmpdir = tempfile.mkdtemp(prefix=TMPDIR_PREFIX)\n+                self._start_logging()\n+                self.log.error(\"The tmpdir must not already exist {}\".format(tmpdirpath))",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#discussion_r1522713384",
      "id" : 1522713384,
      "line" : null,
      "node_id" : "PRRC_kwDOABII585awsMo",
      "original_commit_id" : "09067cc39d1033adc44e3fb1cfeecc2a839e3e97",
      "original_line" : 272,
      "original_position" : 22,
      "original_start_line" : null,
      "path" : "test/functional/test_framework/test_framework.py",
      "position" : null,
      "pull_request_review_id" : 1933396057,
      "pull_request_url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/29498",
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1522713384/reactions"
      },
      "side" : "RIGHT",
      "start_line" : null,
      "start_side" : null,
      "subject_type" : "line",
      "updated_at" : "2024-03-13T08:07:58Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1522713384",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/6399679?v=4",
         "events_url" : "https://api.github.com/users/maflcko/events{/privacy}",
         "followers_url" : "https://api.github.com/users/maflcko/followers",
         "following_url" : "https://api.github.com/users/maflcko/following{/other_user}",
         "gists_url" : "https://api.github.com/users/maflcko/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/maflcko",
         "id" : 6399679,
         "login" : "maflcko",
         "node_id" : "MDQ6VXNlcjYzOTk2Nzk=",
         "organizations_url" : "https://api.github.com/users/maflcko/orgs",
         "received_events_url" : "https://api.github.com/users/maflcko/received_events",
         "repos_url" : "https://api.github.com/users/maflcko/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/maflcko/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/maflcko/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/maflcko"
      }
   },
   {
      "_links" : {
         "html" : {
            "href" : "https://github.com/bitcoin/bitcoin/pull/29498#discussion_r1523392126"
         },
         "pull_request" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/29498"
         },
         "self" : {
            "href" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1523392126"
         }
      },
      "author_association" : "CONTRIBUTOR",
      "body" : "yup that makes sense and I agree reverting back to https://github.com/bitcoin/bitcoin/commit/d4e36ae80d4b3f03647fd9057461edf7ecd794a3",
      "commit_id" : "d4e36ae80d4b3f03647fd9057461edf7ecd794a3",
      "created_at" : "2024-03-13T14:43:49Z",
      "diff_hunk" : "@@ -262,8 +262,15 @@ def setup(self):\n \n         # Set up temp directory and start logging\n         if self.options.tmpdir:\n-            self.options.tmpdir = os.path.abspath(self.options.tmpdir)\n-            os.makedirs(self.options.tmpdir, exist_ok=False)\n+            tmpdirpath = self.options.tmpdir\n+            try:\n+                self.options.tmpdir = os.path.abspath(tmpdirpath)\n+                os.makedirs(self.options.tmpdir, exist_ok=False)\n+            except Exception:\n+                self.options.tmpdir = tempfile.mkdtemp(prefix=TMPDIR_PREFIX)\n+                self._start_logging()\n+                self.log.error(\"The tmpdir must not already exist {}\".format(tmpdirpath))",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/29498#discussion_r1523392126",
      "id" : 1523392126,
      "in_reply_to_id" : 1522713384,
      "line" : null,
      "node_id" : "PRRC_kwDOABII585azR5-",
      "original_commit_id" : "09067cc39d1033adc44e3fb1cfeecc2a839e3e97",
      "original_line" : 272,
      "original_position" : 22,
      "original_start_line" : null,
      "path" : "test/functional/test_framework/test_framework.py",
      "position" : null,
      "pull_request_review_id" : 1934383287,
      "pull_request_url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/29498",
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1523392126/reactions"
      },
      "side" : "RIGHT",
      "start_line" : null,
      "start_side" : null,
      "subject_type" : "line",
      "updated_at" : "2024-03-13T14:43:49Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/1523392126",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/15950706?v=4",
         "events_url" : "https://api.github.com/users/kevkevinpal/events{/privacy}",
         "followers_url" : "https://api.github.com/users/kevkevinpal/followers",
         "following_url" : "https://api.github.com/users/kevkevinpal/following{/other_user}",
         "gists_url" : "https://api.github.com/users/kevkevinpal/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/kevkevinpal",
         "id" : 15950706,
         "login" : "kevkevinpal",
         "node_id" : "MDQ6VXNlcjE1OTUwNzA2",
         "organizations_url" : "https://api.github.com/users/kevkevinpal/orgs",
         "received_events_url" : "https://api.github.com/users/kevkevinpal/received_events",
         "repos_url" : "https://api.github.com/users/kevkevinpal/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/kevkevinpal/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/kevkevinpal/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/kevkevinpal"
      }
   }
]
